node_modules/col-whiteboards/lib/api.js

```diff
@@ -0,0 +1,771 @@
+/**
+ * Copyright 2015 UC Berkeley (UCB) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('lodash');
+var config = require('config');
+var cookie = require('cookie');
+var cookieParser = require('cookie-parser');
+var Joi = require('joi');
+var url = require('url');
+
+var Collabosphere = require('col-core');
+var CollabosphereUtil = require('col-core/lib/util');
+var DB = require('col-core/lib/db');
+var log = require('col-core/lib/logger')('col-whiteboards');
+var UserConstants = require('col-users/lib/constants');
+
+/**
+ * Get a whiteboard
+ * // TODO: Only load online people and whiteboard content when required
+ *
+ * @param  {Number}         userId                          The id of the user retrieving the whiteboard
+ * @param  {Number}         id                              The id of the whiteboard
+ * @param  {Function}       callback                        Standard callback function
+ * @param  {Object}         callback.err                    An error that occurred, if any
+ * @param  {Whiteboard}     callback.whiteboard             The requested whiteboard
+ */
+var getWhiteboard = module.exports.getWhiteboard = function(userId, id, callback) {
```

Comment:
When a new socket connection comes in through `Collabosphere.appServer.io.on('connection'` can we not do something like:
```javascript
Collabosphere.appServer.io.on('connection', function(socket) {
  // Cookie stuff
  var userId = cookieParser.signedCookie(sessionCookie, config.get('cookie.secret'));

  // Referer stuff
  var whiteboardId = referer.pathname.split('/').pop();

  // Get the user
  UsersAPI.getUser(userId, function(err, user) {
    if (err) { return callback(err); }
    // Keep track of the context
    socket.ctx = {'user': user, 'course': user.course};

    // Get whiteboard
    getWhiteboard(socket.ctx, whiteboardId, function(err, whiteboard) {
       ....
    });
  });
};
```
get the user object from which we can generate a context? That would allow us to keep our usual style of passing in a ctx as the first parameter of each API method.